Draft feature added to the pages.

MansurBesleney opened the pull request:
### Description

Resolves #23616 

- User can save a page as draft and publish it later.

### Checklist

- [x] I fully tested it as developer.
- [x] No need to document.

MansurBesleney committed:
Add PageStatus support to Page entities and repositories

Introduces the PageStatus enum and adds a Status property to Page and PageCacheItem. Updates PageManager, IPageRepository, EfCorePageRepository, and MongoPageRepository to support filtering and managing pages by status. This enables handling draft and published states for pages.

MansurBesleney committed:
Add PageStatus support to page DTOs and services

Introduces the PageStatus property to page-related DTOs, allowing filtering and setting of page status in admin APIs. Updates service methods and repository calls to handle the new status parameter, and enforces the status field as required in the EF Core model configuration.

MansurBesleney committed:
Add page status field to create and update forms

Introduces a 'Status' field to both the page creation and update forms in the admin web module, allowing users to set or modify the status of a page. Updates the view models and UI to support this new property.

MansurBesleney committed:
Add page status check for public page access

Introduces a PageStatus property to PageDto and updates PagePublicAppService to only return pages with a published status. This ensures that only published pages are accessible via public endpoints.

MansurBesleney committed:
Add status field to Pages and update migrations

Introduced a new 'Status' field to the Pages entity via EF Core migration. Added migration files and updated the model snapshot to reflect the schema change. Also removed the solution file 'cms-kit.sln'.

MansurBesleney committed:
Add Save as Draft and Publish actions to page forms

Replaces the single submit button with separate 'Save as Draft' and 'Publish' buttons on both Create and Update page forms. Updates the status field handling and form submission logic in the corresponding JavaScript files. Marks the Status property as hidden input in the page models to support the new workflow.

MansurBesleney committed:
proxies regenerated

MansurBesleney committed:
Add page status column to CMS pages list

Introduced a new 'Status' column in the CMS pages admin table, displaying localized page status values. Added corresponding English localization entries for page status enums.

MansurBesleney committed:
Merge branch 'dev' into issue-#23616

gizemmutukurt reviewed (changes requested):
- [ ] PR also receives an error regarding DB.

<img width="2476" height="1418" alt="image" src="https://github.com/user-attachments/assets/be6683f9-18a3-45f6-889e-dcc4aa6e18fc" />

MansurBesleney committed:
regenerate proxies

MansurBesleney committed:
Merge branch 'dev' into issue-#23616

MansurBesleney committed:
Merge branch 'dev' into issue-#23616

MansurBesleney committed:
Make page slug filtering case-insensitive

Updated both EF Core and MongoDB page repositories to perform case-insensitive filtering on the Slug property. Also set the default thread culture to InvariantCulture in tests to avoid culture-specific issues, and added PageStatus to seeded test data for consistency.

MansurBesleney committed:
Pin codemirror to v5.65.1 via resolutions

Added a 'resolutions' field in package.json to enforce codemirror v5.65.1, replacing v6.x dependencies in yarn.lock. This ensures compatibility with packages expecting codemirror v5.
